### PR TITLE
twitter updated to use request

### PR DESCRIPTION
**Pull request by decobots**

### CODE REVIEW

File: twitter.py [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
+import collections
 import json
 
-import requests
 from requests_oauthlib import OAuth1
 
 from environment_variables import get_env
+from request import request
+
+endpoint = collections.namedtuple('twitter_method', ["url", "type"])
```
Comment (livercat) [inline twitter.py]: twitter_method -> endpoint (or endpoint -> twitter_method). Class name should preferably match the name in code.

File: twitter.py [hunk 2]
```diff
@@ -1,59 +1,62 @@
...
-                                   auth=self.auth_post)
-            # format required for delete_tweet_by_id method
-        elif self.method_type[method_name] == "GET":
-            result = requests.get(self.method[method_name], params=kwargs, auth=self.auth_get)
-        return result
-
     def upload_photo(self, name, data):
         data = base64.b64encode(data)
-        upload_photo_result = self.request("twitter_upload_pic", name=name, media_data=data)
+        payload = {"name": name, "media_data": data}
```
Comment (livercat) [inline twitter.py]: No need to create a separate var here, just pass dict directly to "payload" in request()

File: twitter.py [hunk 3]
```diff
@@ -1,59 +1,62 @@
...
+        upload_photo_result = request(self.twitter_upload_pic.type,
+                                      self.twitter_upload_pic.url,
+                                      payload=payload,
+                                      auth=self.auth)
         return json.loads(upload_photo_result.text)['media_id'], name
 
     def create_post(self, status, id_of_photo=[]):
-        self.request("twitter_create_post", status=status, media_ids=id_of_photo)
+        payload = {"status": status,
```
Comment (livercat) [inline twitter.py]: Same - no need for a var

File: twitter.py [hunk 4]
```diff
@@ -1,59 +1,62 @@
...
+                               payload=payload,
+                               auth=self.auth)
+        return json.loads(created_post.content)["id"]
 
     def get_users_posts(self, amount):
-        get_users_posts_request = self.request("twitter_get_users_posts", count=amount)
+        payload = {"count": amount}
+        get_users_posts_request = request(self.twitter_get_users_posts.type,
+                                          self.twitter_get_users_posts.url,
+                                          payload=payload,
```
Comment (livercat) [inline twitter.py]: Same

File: twitter.py (hunk 1 above)
Comment (decobots) [reply twitter.py]: also fixed in flickr

File: twitter.py (hunk 2 above)
Comment (decobots) [reply twitter.py]: also fixed in flickr

**Pull request merged**